plugins/openapi.yaml

```diff
@@ -0,0 +1,496 @@
+              $ref: '#/components/schemas/Stop'
+
+components:
+  schemas:
+
+    Annotations:
+      type: object
+      description: Generic map of string keys and string values to attach arbitrary non-identifying metadata to objects.
+      additionalProperties:
+        type: string
+
+    Metadata:
+      type: object
+      description: Describes the scanner plugin.
+      properties:
+        annotations:
+          $ref: '#/components/schemas/Annotations'
+        name:
+          type: string
+        version:
+          type: string
+        apiVersion:
+          type: string
+
+    Status:
+      type: object
+      required:
+        - state
+        - lastTransitionTime
+      properties:
+        state:
+          description: |
+            Describes the status of scanner.
+            | Status         | Description                                                   |
+            | -------------- | ------------------------------------------------------------- |
+            | NotReady       | Initial state when the scanner container starts               |
+            | Ready          | Scanner setup is complete and it is ready to receive requests |
+            | Running        | Scanner config was received and the scanner is running        |
+            | Failed         | Scanner failed                                                |
+            | Done           | Scanner is completed successfully                             |
+          type: string
+          enum:
+            - NotReady
+            - Ready
+            - Running
+            - Failed
+            - Done
+        message:
+          description: Human readable message.
+          type: string
+        lastTransitionTime:
+          description: Last date time when the status has changed.
+          type: string
+          format: date-time
+
+    Config:
+      type: object
+      description: Describes a scanner config.
+      required:
+        - inputDir
+        - outputFile
+        - timeoutSeconds
+      properties:
+        file:
+          type: string
+          description: |
+            The file with the configuration required by the scanner plugin. This is a path on the filesystem to the config file.
+        inputDir:
+          type: string
+          x-oapi-codegen-extra-tags:
+            validate: required
+          description: |
+            The directory which should be scanned by the scanner plugin.
+        outputFile:
+          type: string
+          x-oapi-codegen-extra-tags:
+            validate: required
+          description: |
+            Path to JSON file where the scanner plugin should store its results.
+        timeoutSeconds:
+          type: integer
+          minimum: 0
+          x-oapi-codegen-extra-tags:
+            validate: required,gt=0
+          description: |
+            The maximum time in seconds that a scan started from this scan
+            should run for before being automatically aborted.
```

Comment:
is the `outputType` property missing here?

Reply:
Yea, I decided to ditch it for now since I couldn't decide on the interface. Let's cover the basic functionality for now. We can abuse the annotations if we need quick way to support custom data (and ingestion)